src/components/DonationCards/DonationCards.tsx

```diff
@@ -18,6 +19,13 @@ function DonationCards() {
                title={donationCard.title}
                description={donationCard.description}
                donationTarget={donationCard.donationTarget}
+               amount={Math.random() * donationCard.donationTarget}
+               isOpen={openCardId === donationCard.id}
+               onShowMore={() =>
+                  setOpenCardId(
+                     openCardId !== donationCard.id ? donationCard.id : -1
+                  )
```

Comment:
Another great place for destructuring so there are no tons of ```donationCard.``` (in the argument of the map).